src/scripts/mod.rs

```diff
@@ -0,0 +1,335 @@
+
+    // Tags found in ScriptFile.
+    tags: Option<Vec<String>>,
+
+    // The format how we want the script to run.
+    call_format: Option<String>,
+}
+
+#[derive(Serialize)]
+struct ExecPartsScript {
+    script: String,
+    ip: String,
+    port: String,
+}
+
+#[derive(Serialize)]
+struct ExecParts {
+    ip: String,
+    port: String,
+}
+
+impl Script {
+    pub fn build(
+        path: Option<PathBuf>,
+        ip: IpAddr,
+        open_ports : Vec<u16>,
+        port: Option<String>,
+        ports_separator: Option<String>,
+        tags: Option<Vec<String>>,
+        call_format: Option<String>,
+    ) -> Self {
+        Self {
+            path: path,
+            ip: ip,
+            open_ports : open_ports ,
+            port: port,
+            ports_separator: ports_separator,
+            tags: tags,
+            call_format: call_format,
+        }
+    }
+
+    // Some variables get chnaged before read, and compiler throws warning on warn(unused_assignments)
+    #[allow(unused_assignments)]
+    pub fn run(self) -> Result<String> {
+        debug!("run self {:?}", &self);
+
+        let mut ports_str = String::new();
+        if let Some(ports_separator) = self.ports_separator {
+            let ports_str_vec: Vec<String> =
+                self.open_ports .iter().map(|port| port.to_string()).collect();
+            ports_str = ports_str_vec.join(&ports_separator);
+        } else if let None = self.ports_separator {
+            let ports_str_vec: Vec<String> =
+                self.open_ports .iter().map(|port| port.to_string()).collect();
+            ports_str = ports_str_vec.join(",");
+        } else if let Some(port) = self.port {
+            ports_str = port;
+        } else {
+            return Err(anyhow!("Failed to parse ports_separator in scriptfile."));
+        }
+
+        let mut final_call_format = String::new();
+        if let Some(call_format) = self.call_format {
+            final_call_format = call_format;
+        } else {
+            return Err(anyhow!("Failed to parse execution format."));
+        }
+        let default_template: Template = Template::new(&final_call_format);
+        let mut to_run = String::new();
+
+        if final_call_format.contains("{{script}}") {
+            let exec_parts_script: ExecPartsScript = ExecPartsScript {
+                script: self.path.unwrap().to_str().unwrap().to_string(),
+                ip: self.ip.to_string(),
+                port: ports_str,
+            };
+            to_run = default_template.fill_with_struct(&exec_parts_script)?;
+        } else {
+            let exec_parts: ExecParts = ExecParts {
+                ip: self.ip.to_string(),
+                port: ports_str,
+            };
+            to_run = default_template.fill_with_struct(&exec_parts)?;
+        }
+
+        debug!("\nTo run {}", to_run);
+
+        let arguments: Vec<String> = to_run.split(" ").map(|arg| arg.to_string()).collect();
```

Comment:
We should probably use `shell_words::split` here instead of a normal split by whitespace to avoid edge cases of argument parsing.

Reply:
I changed to use the original `shell_words::split` parser for the commandline args.